packages/ng-schematics/src/utils/theme-import.ts

```diff
@@ -53,78 +53,52 @@ function importDefaultThemeSass(tree: Tree, ext: string): Tree {
 @include igx-core();
 @include igx-theme($default-palette);
 `;
-	const sourceRoot = getDefaultProject(tree).sourceRoot;
-	const targetFile = `/${sourceRoot}/styles.${ext}`;
-	let content = tree.read(targetFile).toString();
+
+	let content = tree.read(filePath)!.toString();
 
 	if (!content.includes(sassImports)) {
 		content += sassImports;
 	}
 
-	tree.overwrite(targetFile, content);
-	return tree;
+	tree.overwrite(filePath, content);
 }
 
-function importIgDefaultTheme(tree: Tree): Tree {
-	const targetFile = "/angular.json";
-	const workspace = getWorkspace(tree);
-	const importedStylesToBuild = importDefaultThemeToAngularWorkspace(workspace, "build");
-	const importedStylesToTest = importDefaultThemeToAngularWorkspace(workspace, "test");
-
-	if (importedStylesToBuild || importedStylesToTest) {
-		tree.overwrite(targetFile, JSON.stringify(workspace, null, 2) + "\n");
-	}
-
-	return tree;
-}
-
-function getTargetedProjectOptions(project: WorkspaceProject<ProjectType>, target: string) {
-	if (project.targets &&
-		project.targets[target] &&
-		project.targets[target].options) {
-		return project.targets[target].options;
-	}
-
-	if (project.architect &&
-		project.architect[target] &&
-		project.architect[target].options) {
-		return project.architect[target].options;
-	}
+async function importIgDefaultTheme(tree: Tree): Promise<void> {
+	const host = createWorkspaceHost(tree);
+	const { workspace } = await workspaces.readWorkspace("/", host);
 
-	throw new SchematicsException(`Cannot determine the project's configuration for: ${target}`);
-}
+	importDefaultThemeToAngularWorkspace(workspace, "build");
+	importDefaultThemeToAngularWorkspace(workspace, "test");
 
-export function getDefaultProject(tree: Tree): WorkspaceProject<ProjectType> {
-	const workspace = getWorkspace(tree);
-	return workspace.projects[workspace.defaultProject];
+	// workspace tracks changes internally:
+	await workspaces.writeWorkspace(workspace, host);
 }
 
-export function getDefaultProjectBuildOptions(tree: Tree) {
-	return getTargetedProjectOptions(getDefaultProject(tree), "build");
+export async function getDefaultProject(tree: Tree): Promise<workspaces.ProjectDefinition> {
```

Comment:
How about we mode this one to a util file where we can access it anywhere we need it. `core/util/Schematics.ts` could be a good place for the time being, no? 

We could also create a function that returns not only the default project but all projects in a workspace, with the default project being one of them, this should prove a tiny bit more robust.